sync_qubes-os.org_repo.sh

```diff
@@ -1,30 +1,41 @@
-#!/bin/sh
+#!/bin/bash --
+set -eu
 
-pushd `dirname $0`
+dir=$(dirname "$0")
+pushd "$dir"
 
 #DRY="-n"
-[ "x$HOST" == "x" ] && HOST=yum.qubes-os.org
-[ "x$HOST_BASEDIR" == "x" ] && HOST_BASEDIR=/pub/qubes/repo/yum
+if [ "x${DRY+q}" = xq ] && [ "x$DRY" != x-n ]; then
+    echo 'Bad value for $DRY (must be -n or unset)'>&2
+    exit 1
+fi
+
+: "${DRY=}"
+
+[ "x${HOST-}" == "x" ] && HOST=yum.qubes-os.org
+[ "x${HOST_BASEDIR-}" == "x" ] && HOST_BASEDIR=/pub/qubes/repo/yum
 if [ -n "$1" ]; then
-    RELS_TO_SYNC=`basename "$1"`
+    RELS_TO_SYNC=$(basename "$1")
 else
-    RELS_TO_SYNC="`readlink current-release|tr -d /`"
+    echo 'Release must be specified'>&2
+    exit 1
 fi
+[[ "$RELS_TO_SYNC" =~ ^r[1-9]([.-][0-9a-z]*)?$ ]] || {
+    echo 'Bad rel'>&2
+    exit 1
+}
 REPOS_TO_SYNC="current current-testing security-testing"
 
+export PATH=/usr/bin:/bin
+set -x
 for rel in $RELS_TO_SYNC; do
 
     rsync_args=
     for repo in $REPOS_TO_SYNC; do
         rsync_args="$rsync_args $rel/$repo"
     done
     echo "Syncing $rel..."
-    rsync $DRY --partial --progress --hard-links --exclude repodata -air $rsync_args $HOST:$HOST_BASEDIR/$rel/
-    rsync $DRY update_repo.sh update_repo-arg.sh mirrors.list $HOST:
-    for repo in $REPOS_TO_SYNC; do
-        [ -z "$DRY" ] && ssh $HOST sh -c "\"cd $HOST_BASEDIR; ~/update_repo-arg.sh $rel/$repo/dom0/fc* $rel/$repo/vm/*\""
-    done
```

Comment:
Here we'll end up without metalink updates at all - you removed server-side generate, but also abandoned installing the generator on the build VM side (https://github.com/QubesOS/qubes-infrastructure/pull/19).
IMO for now you can simply restore the above server-side generate (adjust args to `update_repo-arg.sh`).

Reply:
Fixed by re-implementing server-side generation.